zrml/hybrid-router/src/lib.rs

```diff
@@ -0,0 +1,634 @@
+            ensure!(
+                price_limit <= ZeitgeistBase::<BalanceOf<T>>::get()?,
+                Error::<T>::PriceLimitTooHigh
+            );
+            let orders: OrdersOf<T> =
+                orders.try_into().map_err(|_| Error::<T>::MaxOrdersExceeded)?;
+            let market = T::MarketCommons::market(&market_id)?;
+            let assets = Self::outcome_assets(market_id, &market);
+            ensure!(asset_count as usize == assets.len(), Error::<T>::AssetCountMismatch);
+
+            let (asset_in, asset_out) = match tx_type {
+                TxType::Buy => (market.base_asset, asset),
+                TxType::Sell => (asset, market.base_asset),
+            };
+            T::AssetManager::ensure_can_withdraw(asset_in, &who, amount_in)?;
+
+            let mut remaining = amount_in;
+
+            remaining = Self::maybe_fill_orders(
+                tx_type,
+                &orders,
+                remaining,
+                &who,
+                market_id,
+                market.base_asset,
+                asset,
+                price_limit,
+            )?;
+
+            if !remaining.is_zero() {
+                remaining = Self::maybe_fill_from_amm(
+                    tx_type,
+                    &who,
+                    market_id,
+                    asset,
+                    remaining,
+                    price_limit,
+                )?;
+            }
+
+            if !remaining.is_zero() {
+                let (maker_asset, maker_amount, taker_asset, taker_amount) = match tx_type {
+                    TxType::Buy => {
+                        let maker_asset = market.base_asset;
+                        let maker_amount = remaining;
+                        let taker_asset = asset;
+                        let taker_amount = remaining.bdiv_ceil(price_limit)?;
+                        (maker_asset, maker_amount, taker_asset, taker_amount)
+                    }
+                    TxType::Sell => {
+                        let maker_asset = asset;
+                        let maker_amount = remaining;
+                        let taker_asset = market.base_asset;
+                        let taker_amount = price_limit.bmul_floor(remaining)?;
+                        (maker_asset, maker_amount, taker_asset, taker_amount)
+                    }
+                };
+
+                Self::maybe_place_limit_order(
+                    strategy,
+                    &who,
+                    market_id,
+                    maker_asset,
+                    maker_amount,
+                    taker_asset,
+                    taker_amount,
+                )?;
+            }
+
+            Self::deposit_event(Event::HybridRouterExecuted {
+                tx_type,
+                who,
+                market_id,
+                price_limit,
+                asset_in,
+                amount_in,
+                asset_out,
+            });
```

Comment:
This event will have additional fields in the AMM-CDA-7 PR.

Reply:
For more information look here https://github.com/zeitgeistpm/zeitgeist/pull/1288